subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/RelevantProjectsRegistry.kt

```diff
@@ -31,7 +31,7 @@ class RelevantProjectsRegistry : ProjectAccessHandler {
     fun relevantProjects(nodes: List<Node>): Set<ProjectInternal> =
         targetProjects +
             nodes.mapNotNullTo(mutableListOf()) { node ->
-                node.owningProject?.takeIf { it.parent != null }
```

Comment:
@eskatos was there a reason for this? All tests pass and it fixes a bug where the build directory of the root project would not be restored.